File: vocal.pas
program vocal;

var infile:text;   (* Input file *)
    ch:char;       (* Current character from input file *)

begin
   (* Open the input file for reading *)
   assign(infile, 'vocal.in');
   reset(infile);

   (* Process the file until the end of it *)
   while not eof(infile) do
   begin

      (* For each character in the line, we want to read and process it *)
      while not eoln(infile) do
      begin

         (* Read a character from the input file *)
         read(infile, ch);

         (* If the character is one of the special characters that should *)
         (* be changed to a noise, output the appropriate noise string    *)
         case ch of
            '.': write('[pthut]');
            '!': write('[phsss-pthut]');
            ',': write('[pst]');
            '-': write('[fshhh]');
            '''': write('[chic]');
            '"': write('[chic-chic]');
            '?': write('[pssu-pthut]');
         else
            (* Otherwise, output the character unchanged *)
            write(ch);
         end;
      end;

      (* At end of line, tell the input file to go to next line and *)
      (* write out a CR/LF to go to the next line in the output     *)
      readln(infile);
      writeln;
   end;

   (* Close the input file *)
   close(infile);
end.

File: vocal.c
#include <stdio.h>

void main(void)
{
   FILE   *infile;   /* Input file pointer */
   char   ch;        /* Current character in input file */

   /* Assume the input file exists and open it for reading */
   infile = fopen("vocal.in", "r");

   /* Process file until end of it */
   while (!feof(infile))
   {
      /* Get the first character */
      ch = fgetc(infile);

      /* Keep looping until either end-of-file or end-of-line (newline) */
      while ( (!feof(infile)) && (ch != '\n') )
      {
	 /* If the current character is one of the special ones needed to */
	 /* be converted to a sound, output the sound representation.     */
	 /* Otherwise, just output the character unchanged.               */
	 switch (ch)
	 {
	    case '.':
	       printf("[pthut]");
	       break;
	    case '!':
	       printf("[phsss-pthut]");
	       break;
	    case ',':
	       printf("[pst]");
	       break;
	    case '-':
	       printf("[fshhh]");
	       break;
	    case '\'':
	       printf("[chic]");
	       break;
	    case '"':
	       printf("[chic-chic]");
	       break;
	    case '?':
	       printf("[pssu-pthut]");
	       break;
	    default:
	       printf("%c", ch);
	       break;
	 }

	 /* Get the next character */
	 ch = fgetc(infile);
      }

      /* We reached end-of-line so write a newline on the output */
      printf("\n");
   }

   /* Close the input file */
   fclose(infile);
}